android/src/main/kotlin/com/wire/kalium/presentation/MainActivity.kt

```diff
@@ -1,65 +1,65 @@
 package com.wire.kalium.presentation
 
+import android.content.Context
+import android.os.Build
 import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
 import androidx.compose.foundation.layout.Column
 import androidx.compose.material.Text
 import androidx.compose.runtime.Composable
-import com.wire.kalium.cryptography.CryptoClientId
-import com.wire.kalium.cryptography.CryptoSessionId
-import com.wire.kalium.cryptography.ProteusClient
-import com.wire.kalium.cryptography.UserId
-import java.io.File
+import com.wire.kalium.logic.CoreLogic
+import com.wire.kalium.logic.data.conversation.Conversation
+import com.wire.kalium.logic.feature.auth.LoginUsingEmailUseCase
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.runBlocking
 
 class MainActivity : ComponentActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        val rootProteusDir = File(this.filesDir, "proteus")
-
-        val alice = SampleUser("aliceId", "Alice")
-        val aliceFile = File(rootProteusDir, alice.id)
-        aliceFile.mkdirs()
-        val aliceClient = ProteusClient(rootProteusDir.absolutePath, alice.id)
-        aliceClient.open()
-        val aliceSessionId = CryptoSessionId(UserId(alice.id), CryptoClientId("aliceClient"))
-        val aliceKey = aliceClient.newPreKeys(0, 10).first()
-
-        val bob = SampleUser("bobId", "Bob")
-        val bobFile = File(rootProteusDir, bob.id)
-        bobFile.mkdirs()
-        val bobClient = ProteusClient(rootProteusDir.absolutePath, bob.id)
-        val bobSessionId = CryptoSessionId(UserId(bob.id), CryptoClientId("bobClient"))
-        bobClient.open()
+        setContent {
+            val conversations = runBlocking {
+                val rootProteusDirectoryPath = getDir("proteus", Context.MODE_PRIVATE).path
+                val core = CoreLogic(clientLabel, rootProteusDirectoryPath)
```

Comment:
I know we are still in exploratory phase, but should we move this core logic singleton instance creation to the `AndroidApplication` level so that it´s visible everywhere?

Reply:
I'll change it to go this way.

Maybe later if we plug some DI into this, we can change it again.

Reply:
@MohamadJaara is doing this in the sample app already, so I'll leave it to him 🙈